MatrixE2ETest: Popover-Test schlägt im Suite-Lauf fehl (geteilte Testdaten, .first()-Zelle)

## Problem

`MatrixE2ETest.cell_detail_popover_shows_the_complete_task_description` schlägt fehl, sobald die E2E-Suite als Ganzes in einer JVM läuft. Einzeln aufgerufen ist der Test grün.

```
[ERROR] MatrixE2ETest.cell_detail_popover_shows_the_complete_task_description
org.opentest4j.AssertionFailedError:
Locator expected to contain text
Expected: Sehr ausfuehrliche Beschreibung der Taetigkeit, die deutlich breiter ist als das Popover …
Received: 2:00: E2E-Testbuchung
```

Reproduzierbar mit:

```bash
jenv exec ./mvnw -o test -Dtest='DailyBookingE2ETest,MatrixE2ETest' -De2e.browsers=chrome
```

**Nicht** von einer Produktivänderung verursacht — gegen den unveränderten `main`-Stand (per `git stash` verifiziert) tritt der Fehler identisch auf.

## Ursache

Die E2E-Tests teilen sich eine H2-Datenbank pro Testlauf. `E2ETestData.seedIfNeeded` legt die Stammdaten einmal an, die von den Tests **erzeugten Buchungen bleiben aber über Testklassen hinweg stehen**. Beide Tests buchen als `ema` in denselben Monat (Juni 2026):

| Test | Datum | Unterauftrag | Kommentar |
|---|---|---|---|
| `DailyBookingE2ETest` | 2026-06-15 | `ALPHA-DEV` | `E2E-Testbuchung` |
| `MatrixE2ETest` | 2026-06-17 | `GLOBEX-CONSULT` | der lange Text |

Der Test greift die Zelle über `.first()` (`MatrixE2ETest.java:66`):

```java
page.locator("#matrix tbody td:has(.matrix-cell-detail)").first().hover();
```

Sobald `DailyBookingE2ETest` vorher gelaufen ist, existiert in der Matrix eine weitere Zeile mit Detailzelle. Welche davon `.first()` trifft, hängt an der Zeilen- und Spaltensortierung der Matrix, nicht am Test — getroffen wird die Buchung der anderen Testklasse, und die Assertion prüft anschließend den falschen Popover.

Die Assertion selbst ist in Ordnung: sie sichert die Regression aus #818 ab (Detailzeilen waren `<pre>`-Blöcke und wurden abgeschnitten). Nur ihr Ziel ist unspezifisch ausgewählt.

## Lösungsvorschlag

Der billigste und zugleich robusteste Fix ist, die Zelle gezielt zu adressieren statt `.first()` zu nehmen — der Test kennt seinen Unterauftrag ja:

```java
page.locator("#matrix tbody tr")
    .filter(new Locator.FilterOptions().setHasText(E2ETestData.SUBORDER_GLOBEX_CONSULT_SIGN))
    .locator("td:has(.matrix-cell-detail)")
    .first()
    .hover();
```

Damit ist der Test unabhängig davon, was andere Testklassen in denselben Monat gebucht haben.

Weitergehende Alternativen, falls das Muster häufiger auftritt:

- **Eigene Zeitscheibe pro Testklasse:** jede E2E-Klasse bucht in einen eigenen Monat, damit sich Matrix- und Monatsansichten nicht überlagern. Verhindert die ganze Fehlerklasse, kostet aber eine Konvention, an die man sich erinnern muss.
- **Buchungen nach jeder Testklasse aufräumen** (`@AfterAll`). Am saubersten, aber der teuerste Eingriff in `PlaywrightE2ETestBase`, und die Stammdaten sollen bewusst erhalten bleiben.

Empfehlung: den gezielten Locator umsetzen, und die Zeitscheiben-Konvention als Notiz in `PlaywrightE2ETestBase` festhalten, damit der nächste Test nicht in dieselbe Falle läuft.

## Akzeptanzkriterien

- [ ] `jenv exec ./mvnw -o test -Dtest='org.tb.e2e.**' -De2e.browsers=chrome` läuft vollständig grün
- [ ] `MatrixE2ETest` ist auch einzeln weiterhin grün
- [ ] Der Test adressiert die von ihm selbst erzeugte Buchung, nicht die erste Detailzelle der Matrix
- [ ] Die Absicherung aus #818 (Umbruch im Popover statt Abschneiden) bleibt erhalten
- [ ] Grün in beiden Browsern der Matrix (`-De2e.browsers=chrome` und `firefox`)

--- a/src/test/java/org/tb/e2e/PlaywrightE2ETestBase.java
+++ b/src/test/java/org/tb/e2e/PlaywrightE2ETestBase.java
@@ -43,6 +43,18 @@ import org.tb.order.persistence.SuborderRepository;
  * ({@code local} profile, see {@link org.tb.auth.configuration.LocalDevSecurityConfiguration}).
  * Master data (customers, orders, employees, ...) is seeded once per test run via
  * {@link E2ETestData}.
+ *
+ * <p><b>Shared database state:</b> all E2E test classes run against the same H2 database, and
+ * the bookings a test creates are never cleaned up — they stay visible to every test class that
+ * runs afterwards, and a {@code @ParameterizedTest} over {@link #browsers()} creates them once
+ * per browser. Tests must therefore assert on the data they created themselves instead of on
+ * "the first row/cell" of a view (#846). Two conventions keep that manageable:
+ * <ul>
+ *   <li>address elements by something unique to the test (its suborder, its comment text)
+ *       rather than by position,</li>
+ *   <li>prefer a dedicated day or month per test class, so month-spanning views such as the
+ *       matrix overview do not mix bookings of different classes in the first place.</li>
+ * </ul>
  */
 @SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, properties = "management.health.mail.enabled=false")
 @ActiveProfiles({"unittest", "local"})

--- a/src/test/java/org/tb/e2e/dailyreport/MatrixE2ETest.java
+++ b/src/test/java/org/tb/e2e/dailyreport/MatrixE2ETest.java
@@ -63,15 +63,22 @@ class MatrixE2ETest extends PlaywrightE2ETestBase {
           "/dailyreport/matrix?month=" + BOOKING_DATE.getMonthValue() + "&year=" + BOOKING_DATE.getYear(),
           E2ETestData.EMPLOYEE_MA_SIGN));
 
-      page.locator("#matrix tbody td:has(.matrix-cell-detail)").first().hover();
+      // address the cell this test booked itself: the E2E suite shares one database, so other
+      // test classes leave bookings of their own in the same month (see PlaywrightE2ETestBase)
+      page.locator("#matrix tbody td:has(.matrix-cell-detail)")
+          .filter(new Locator.FilterOptions().setHasText(LONG_COMMENT))
+          .first()
+          .hover();
 
       Locator popover = page.locator(".matrix-detail-popover");
       assertThat(popover).isVisible();
       assertThat(popover).containsText(LONG_COMMENT);
 
       // the description has to wrap inside the popover instead of overflowing it sideways
-      assertEquals(true, popover.locator(".popover-body > *").first()
-          .evaluate("el => el.scrollWidth <= el.clientWidth + 1"));
+      Locator longDetailLine = popover.locator(".popover-body > *")
+          .filter(new Locator.FilterOptions().setHasText(LONG_COMMENT))
+          .first();
+      assertEquals(true, longDetailLine.evaluate("el => el.scrollWidth <= el.clientWidth + 1"));
     });
   }
 

